src/box/lua/net_box.lua

```diff
@@ -450,6 +467,15 @@ local function check_eval_args(args)
     end
 end
 
+local function check_opts_buffer(opts)
+    if opts ~= nil and opts.buffer ~= nil and
```

Comment:
Can you do this check in `check_param_table`? For example, you could pass a callback instead of a type, in which case `check_param_table` would call the callback instead of checking the type directly. The callback could return true on success and false + expected type on error so that error raising would be consolidated in one place - in `check_param_table`. Another alternative is to pass `ffi.ctype` and make `check_param_table` check the type with `ffi.istype`. A bad thing about the latter is that we couldn't use it for checking native `cdata` and `userdata` with this function, but this isn't required for now so this would probably be fine.